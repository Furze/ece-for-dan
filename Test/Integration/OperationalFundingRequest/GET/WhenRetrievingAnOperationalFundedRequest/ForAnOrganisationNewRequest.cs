using System;
using System.Collections.Generic;
using System.Linq;
using Bard;
using Events.Integration;
using Events.Integration.Protobuf.Ece;
using MoE.ECE.Domain.Event.OperationalFunding;
using MoE.ECE.Domain.Model.ValueObject;
using MoE.ECE.Domain.Read.Model.OperationalFunding;
using MoE.ECE.Integration.Tests.Chapter;
using MoE.ECE.Integration.Tests.Infrastructure;
using Shouldly;
using Xunit;
using Xunit.Abstractions;

namespace MoE.ECE.Integration.Tests.OperationalFundingRequest.GET.WhenRetrievingAnOperationalFundedRequest
{
    public class ForAnOrganisationNewRequest : SpeedyIntegrationTestBase
    {
        public ForAnOrganisationNewRequest(RunOnceBeforeAllTests testSetUp, ITestOutputHelper output,
            TestState<ECEStoryBook, ECEStoryData> testState) : base(testSetUp, output, testState)
        {
        }

        private Guid _businessEntityId;

        protected override void Arrange() =>
            Given
                .A_rs7_skeleton_has_been_created(rs7 =>
                {
                    rs7.FundingPeriod = FundingPeriodMonth.March;
                    rs7.FundingPeriodYear = 2020;
                })
                .An_rs7_is_ready_for_internal_ministry_review()
                .GetResult(data => _businessEntityId = data.Rs7Model.BusinessEntityId.GetValueOrDefault());

        protected override void Act() =>
            When.Get(
                $"api/operational-funding-requests?business-entity-id={_businessEntityId}&revision-number={1}");

        [Fact]
        public void ThenADomainEventShouldBePublished() =>
            A_domain_event_should_be_fired<OperationalFundingRequestCreated>();

        [Fact]
        public void ThenAnIntegrationEventShouldBeCalculated()
        {
            var integrationEvent = An_integration_event_should_be_fired<EntitlementCalculated>();

            integrationEvent
                .BusinessEntityType
                .ShouldBe(Constants.BusinessEntityTypes.Rs7);
        }

        [Fact]
        public void ThenTheResponseShouldBeOk() => Then.Response.ShouldBe.Ok();

        [Fact]
        public void ThenTheResponseShouldContain1Item()
        {
            var result =
                Then.Response.Content<ICollection<OperationalFundingRequestModel>>();

            result.Count.ShouldBe(1);
        }

        [Fact]
        public void ThenTheResponseShouldContainMatchingAdvanceMonths()
        {
            var result =
                Then.Response.Content<ICollection<OperationalFundingRequestModel>>();

            result.ShouldNotBeNull();
            result.ShouldNotBeEmpty();
            result.ElementAt(0).MatchingAdvanceMonths.ShouldNotBeNull();
            result.ElementAt(0).MatchingAdvanceMonths?.Count.ShouldBe(4);
            result.ElementAt(0).MatchingAdvanceMonths?.ElementAt(0).MonthNumber.ShouldBe(10);
            result.ElementAt(0).EntitlementMonths.ShouldNotBeNull();
            result.ElementAt(0).EntitlementMonths.ShouldNotBeEmpty();
            result.ElementAt(0).EntitlementMonths?.ElementAt(0).FundingComponents.ShouldNotBeNull();
            result.ElementAt(0).EntitlementMonths?.ElementAt(0).FundingComponents.ShouldNotBeEmpty();
            result.ElementAt(0).EntitlementMonths?.ElementAt(0).FundingComponents?.ElementAt(0).FundingComponentType
                .ShouldBe(FundingComponent.UnderTwo);
        }

        [Fact]
        public void ThenTheResponseShouldContainTotalForAdvanceMonths()
        {
            var result =
                Then.Response.Content<ICollection<OperationalFundingRequestModel>>();

            result.ShouldNotBeNull();
            result.ShouldNotBeEmpty();
            result.ElementAt(0).MatchingAdvanceMonths.ShouldNotBeNull();

            result.ElementAt(0).AdvanceMonths?.ElementAt(0).TotalAdvance.ShouldBe(418.68m);
            result.ElementAt(0).EntitlementMonths.ShouldNotBeNull();
            result.ElementAt(0).EntitlementMonths?.ElementAt(0).FundingComponents.ShouldNotBeNull();
            result.ElementAt(0).EntitlementMonths?.ElementAt(0).FundingComponents?.ElementAt(0).FundingComponentType
                .ShouldBe(FundingComponent.UnderTwo);
        }
        
        [Fact]
        public void Then_the_response_snapshot_should_be_ok() =>
            Then.Snapshot().Match<ICollection<OperationalFundingRequestModel>>(IgnoreFieldsFor.CollectionOperationalFundingRequestModel);
    }
}