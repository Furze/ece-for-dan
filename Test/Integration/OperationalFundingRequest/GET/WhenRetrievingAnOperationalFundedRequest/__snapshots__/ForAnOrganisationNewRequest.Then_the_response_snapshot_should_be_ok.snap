[
  {
    "EntitlementMonths": [
      {
        "MonthName": "October 2019",
        "MonthNumber": 10,
        "Year": 2019,
        "AllDayCertificatedTeacherHours": 23,
        "AllDayNonCertificatedTeacherHours": 46,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 23,
        "WashUpPlusTen": -325.08,
        "WashUpUnderTwo": -5617.35,
        "WashUpTwoAndOver": -4852.11,
        "WashUpTwentyHours": -7750.05,
        "TotalEntitlement": 6309.36,
        "TotalFundsAdvanced": 24853.95,
        "TotalEntitlementUnderTwo": 3726.0,
        "TotalEntitlementTwoAndOver": 107.64,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": 2475.72,
        "TotalWashUp": -18544.59,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2019-10-01T00:00:00",
            "Amount": 3726.0,
            "Rate": 9.0,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 414,
            "OperatingDays": 23
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2019-10-01T00:00:00",
            "Amount": 107.64,
            "Rate": 4.68,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 23,
            "OperatingDays": 23
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2019-10-01T00:00:00",
            "Amount": 2475.72,
            "Rate": 4.68,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 529,
            "OperatingDays": 23
          }
        ]
      },
      {
        "MonthName": "November 2019",
        "MonthNumber": 11,
        "Year": 2019,
        "AllDayCertificatedTeacherHours": 21,
        "AllDayNonCertificatedTeacherHours": 42,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 21,
        "WashUpPlusTen": -540.36,
        "WashUpUnderTwo": -5941.35,
        "WashUpTwoAndOver": -4861.47,
        "WashUpTwentyHours": -7750.05,
        "TotalEntitlement": 5760.72,
        "TotalFundsAdvanced": 24853.95,
        "TotalEntitlementUnderTwo": 3402.0,
        "TotalEntitlementTwoAndOver": 98.28,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": 2260.44,
        "TotalWashUp": -19093.23,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2019-11-01T00:00:00",
            "Amount": 3402.0,
            "Rate": 9.0,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 378,
            "OperatingDays": 21
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2019-11-01T00:00:00",
            "Amount": 98.28,
            "Rate": 4.68,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 21,
            "OperatingDays": 21
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2019-11-01T00:00:00",
            "Amount": 2260.44,
            "Rate": 4.68,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 483,
            "OperatingDays": 21
          }
        ]
      },
      {
        "MonthName": "December 2019",
        "MonthNumber": 12,
        "Year": 2019,
        "AllDayCertificatedTeacherHours": 22,
        "AllDayNonCertificatedTeacherHours": 44,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 22,
        "WashUpPlusTen": -432.72,
        "WashUpUnderTwo": -5779.35,
        "WashUpTwoAndOver": -4856.79,
        "WashUpTwentyHours": -7750.05,
        "TotalEntitlement": 6035.04,
        "TotalFundsAdvanced": 24853.95,
        "TotalEntitlementUnderTwo": 3564.0,
        "TotalEntitlementTwoAndOver": 102.96,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": 2368.08,
        "TotalWashUp": -18818.91,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2019-12-02T00:00:00",
            "Amount": 3564.0,
            "Rate": 9.0,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 396,
            "OperatingDays": 22
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2019-12-02T00:00:00",
            "Amount": 102.96,
            "Rate": 4.68,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 22,
            "OperatingDays": 22
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2019-12-02T00:00:00",
            "Amount": 2368.08,
            "Rate": 4.68,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 506,
            "OperatingDays": 22
          }
        ]
      },
      {
        "MonthName": "January 2020",
        "MonthNumber": 1,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 23,
        "AllDayNonCertificatedTeacherHours": 46,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 23,
        "WashUpPlusTen": -282.76,
        "WashUpUnderTwo": -5551.11,
        "WashUpTwoAndOver": -4850.27,
        "WashUpTwentyHours": -7750.05,
        "TotalEntitlement": 6419.76,
        "TotalFundsAdvanced": 24853.95,
        "TotalEntitlementUnderTwo": 3792.24,
        "TotalEntitlementTwoAndOver": 109.48,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": 2518.04,
        "TotalWashUp": -18434.19,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-01-01T00:00:00",
            "Amount": 3792.24,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 414,
            "OperatingDays": 23
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-01-01T00:00:00",
            "Amount": 109.48,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 23,
            "OperatingDays": 23
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-01-01T00:00:00",
            "Amount": 2518.04,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 529,
            "OperatingDays": 23
          }
        ]
      }
    ],
    "OrganisationId": 114895,
    "BusinessEntityId": "b12a1131-503d-4bd2-906a-45a8e72b31a4",
    "RequestId": "102001",
    "FundingPeriodMonth": "March",
    "FundingYear": 2020,
    "RevisionNumber": 1,
    "TotalWashUp": -74890.92,
    "TotalAdvance": 1674.72,
    "AdvanceMonths": [
      {
        "MonthName": "March 2020",
        "MonthNumber": 3,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 2,
        "SessionalWorkingDays": null,
        "AmountPayablePlusTen": 164.22,
        "AmountPayableUnderTwo": 247.32,
        "AmountPayableTwoAndOver": 7.14,
        "AmountPayableTwentyHours": 0.0,
        "TotalDays": 2,
        "TotalAdvance": 418.68,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-03-02T00:00:00",
            "Amount": 247.32,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 36,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-03-02T00:00:00",
            "Amount": 7.14,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-03-02T00:00:00",
            "Amount": 164.22,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 46,
            "OperatingDays": 2
          }
        ]
      },
      {
        "MonthName": "April 2020",
        "MonthNumber": 4,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 2,
        "SessionalWorkingDays": null,
        "AmountPayablePlusTen": 164.22,
        "AmountPayableUnderTwo": 247.32,
        "AmountPayableTwoAndOver": 7.14,
        "AmountPayableTwentyHours": 0.0,
        "TotalDays": 2,
        "TotalAdvance": 418.68,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 247.32,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 36,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 7.14,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 164.22,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 46,
            "OperatingDays": 2
          }
        ]
      },
      {
        "MonthName": "May 2020",
        "MonthNumber": 5,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 2,
        "SessionalWorkingDays": null,
        "AmountPayablePlusTen": 164.22,
        "AmountPayableUnderTwo": 247.32,
        "AmountPayableTwoAndOver": 7.14,
        "AmountPayableTwentyHours": 0.0,
        "TotalDays": 2,
        "TotalAdvance": 418.68,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 247.32,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 36,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 7.14,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 164.22,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 46,
            "OperatingDays": 2
          }
        ]
      },
      {
        "MonthName": "June 2020",
        "MonthNumber": 6,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 2,
        "SessionalWorkingDays": null,
        "AmountPayablePlusTen": 164.22,
        "AmountPayableUnderTwo": 247.32,
        "AmountPayableTwoAndOver": 7.14,
        "AmountPayableTwentyHours": 0.0,
        "TotalDays": 2,
        "TotalAdvance": 418.68,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-06-02T00:00:00",
            "Amount": 247.32,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 36,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-06-02T00:00:00",
            "Amount": 7.14,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-06-02T00:00:00",
            "Amount": 164.22,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 46,
            "OperatingDays": 2
          }
        ]
      }
    ],
    "MatchingAdvanceMonths": [
      {
        "MonthName": "October",
        "MonthNumber": 10,
        "Year": 2019,
        "ParentLedWorkingDays": null,
        "AllDayWorkingDays": 20,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 2800.8,
        "AmountPayableUnderTwo": 9343.35,
        "AmountPayableTwoAndOver": 4959.75,
        "AmountPayableTwentyHours": 7750.05,
        "TotalDays": null,
        "TotalAdvance": null,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 2800.8,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 9343.35,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 4959.75,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 7750.05,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          }
        ]
      },
      {
        "MonthName": "November",
        "MonthNumber": 11,
        "Year": 2019,
        "ParentLedWorkingDays": null,
        "AllDayWorkingDays": 20,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 2800.8,
        "AmountPayableUnderTwo": 9343.35,
        "AmountPayableTwoAndOver": 4959.75,
        "AmountPayableTwentyHours": 7750.05,
        "TotalDays": null,
        "TotalAdvance": null,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 2800.8,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 9343.35,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 4959.75,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 7750.05,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          }
        ]
      },
      {
        "MonthName": "December",
        "MonthNumber": 12,
        "Year": 2019,
        "ParentLedWorkingDays": null,
        "AllDayWorkingDays": 20,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 2800.8,
        "AmountPayableUnderTwo": 9343.35,
        "AmountPayableTwoAndOver": 4959.75,
        "AmountPayableTwentyHours": 7750.05,
        "TotalDays": null,
        "TotalAdvance": null,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 2800.8,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 9343.35,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 4959.75,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 7750.05,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          }
        ]
      },
      {
        "MonthName": "January",
        "MonthNumber": 1,
        "Year": 2020,
        "ParentLedWorkingDays": null,
        "AllDayWorkingDays": 20,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 2800.8,
        "AmountPayableUnderTwo": 9343.35,
        "AmountPayableTwoAndOver": 4959.75,
        "AmountPayableTwentyHours": 7750.05,
        "TotalDays": null,
        "TotalAdvance": null,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 2800.8,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 9343.35,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 4959.75,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 7750.05,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          }
        ]
      }
    ],
    "Equity": null
  }
]
